x-pack/plugins/enterprise_search/public/applications/shared/layout/nav_link_helpers.ts

```diff
@@ -11,38 +11,45 @@ import { stripTrailingSlash } from '../../../../common/strip_slashes';
 
 import { KibanaLogic } from '../kibana';
 import { generateReactRouterProps, ReactRouterProps } from '../react_router_helpers';
+import { GeneratedReactRouterProps } from '../react_router_helpers/generate_react_router_props';
 
 interface Params {
   to: string;
-  isRoot?: boolean;
   shouldShowActiveForSubroutes?: boolean;
   items?: Array<EuiSideNavItemType<unknown>>; // Primarily passed if using `items` to determine isSelected - if not, you can just set `items` outside of this helper
 }
 
-export const generateNavLink = ({ to, items, ...rest }: Params & ReactRouterProps) => {
-  return {
-    ...generateReactRouterProps({ to, ...rest }),
-    isSelected: getNavLinkActive({ to, items, ...rest }),
-    items,
+type NavLinkProps = GeneratedReactRouterProps &
+  Pick<EuiSideNavItemType<unknown>, 'isSelected' | 'items'>;
```

Comment:
question: instead of picking from <unknown> can we do something like:
```
type NavLinkProps<T> = GeneratedReactRouterProps & Pick<EuiSideNavItemType<T>>, 'isSelected' | 'items'>;
```